lib/readline.js

```diff
@@ -749,27 +735,13 @@ Interface.prototype._getDisplayPos = function(str) {
   return { cols, rows };
 };
 
-
 // Returns current cursor's position and line
 Interface.prototype.getCursorPos = function() {
-  const columns = this.columns;
   const strBeforeCursor = this._prompt + this.line.substring(0, this.cursor);
-  const dispPos = this._getDisplayPos(strBeforeCursor);
-  let cols = dispPos.cols;
-  let rows = dispPos.rows;
-  // If the cursor is on a full-width character which steps over the line,
-  // move the cursor to the beginning of the next line.
-  if (cols + 1 === columns &&
-      this.cursor < this.line.length &&
-      getStringWidth(this.line[this.cursor]) > 1) {
-    rows++;
-    cols = 0;
-  }
```

Comment:
Our cursor can't be between two code points. We move the cursor by the correct length since a while.